### PR TITLE
Tickable

### PR DESCRIPTION
**Pull request by ronyeh**
This PR fixes https://github.com/0xfe/vexflow/issues/1097

While migrating tests (the one that used a MockTickable), I discovered that the the separation between Note, StaveNote, and Tickable was not very clear. Multiple methods declared that they accepted Tickable, but assumed that the tickables were all Notes or StaveNotes.

This PR makes the separation a bit more clear, and uses `Tickable` instead of `Note`/`StaveNote` where appropriate. I'm sure it can be improved more in the future as we discover more bugs/API-leaks/etc, but I think it's in a good enough state at the moment to be merged into master.

No visual diffs from current master.

### CODE REVIEW

File: src/tickable.ts [hunk 1]
```diff
@@ -324,4 +351,38 @@ export abstract class Tickable extends Element {
...
+
+  /** Get `ModifierContext`. */
+  getModifierContext(): ModifierContext | undefined {
+    return this.modifierContext;
+  }
+
+  /**
+   * This is also used by things like beams and dots for positioning.
+   */
+  isRest(): boolean {
```
Comment (0xfe) [inline src/tickable.ts:376]: What happens if we remove this? It feels semantically out of place.

File: tests/support/qunit_api.ts [hunk 2]
```diff
@@ -5,12 +5,21 @@
 /* eslint-disable */
 
 // global.VF == Vex.Flow and is available in all of the tests.
+// TODO: Remove this after we have cleaned up the VF.* prefixes in the tests.
```
Comment (0xfe) [inline tests/support/qunit_api.ts:8]: I know I merged this in the other PR, however I just realized that the `support` directory is for drop-in third party code.

I'd prefer this to remain in the `tests/` directory -- we can call it `tests/qunit.ts`.

**Review by 0xfe — COMMENTED** [review]
Good change overall. Some quick comments here.

(And we should have a separate discussion about a `Tickable` interface, just concerned about the time, not positioning/rendering, which was the original intent.)

File: tests/support/qunit_api.ts (hunk 2 above)
Comment (ronyeh) [reply tests/support/qunit_api.ts]: OK, will do.

File: src/tickable.ts (hunk 1 above)
Comment (ronyeh) [reply src/tickable.ts:376]: Originally it was in note.ts. I think I pulled it up into Tickable due to some TS warnings, but now that isCategory() is implemented, I think I can move it back into note.ts where it probably belongs!

However, I think the original comment is incorrect. Can you suggest a better comment?

note.ts:
```
  /**
   * Don't play notes by default, call them rests. This is also used by things like
   * beams and dots for positioning.
   */
  isRest(): boolean {
    return false;
  }
```

It says "call them rests" but the actual method returns false, so we aren't actually calling them rests. :-)

Or we can just use a simpler comment like in stavenote.ts:
```
  // Determine if current note is a rest.
  isRest(): boolean {
    return this.glyph.rest;
  }
```

There is another small "bug" that I have noted. this.glyph.rest is actually `true` or `undefined`... JS treats undefined as false in all the places we currently use isRest(), but at some point we should make sure isRest() actually returns a boolean.

**Comment by ronyeh** [comment]
I think I found a good solution for QUnit:

https://github.com/0xfe/vexflow/blob/98d90995d4e5d44ece8fb2ecea2f84b0a04f1f84/tests/types/qunit.d.ts

I moved it into `tests/types/qunit.d.ts` since we also have a `src/types/common.d.ts` with type declarations.

Since `flow.html` provides all the QUnit assertion functions as global functions, I declared them inside `declare global { ... }` which makes the functions available in all test files without having to import the QUnit module. 

This makes the test code slightly closer to what "real code" might look like, since there are now no references to `assert.equal()`, or QUnit imports. The test harness blends into the background and makes the code easier to skim for someone who is learning VexFlow.

No visual diffs from master @ 366057d9d21c710de49c9553e9c4025d885a8615

File: src/tickable.ts (hunk 1 above)
Comment (0xfe) [reply src/tickable.ts]: Okay, let's make this work, and deal with the abstractions after. Re: comment "Returns true if this note is a type of rest. Rests don't have pitches, but take up space in the score."

**Comment by 0xfe** [comment]
Overall, looks good. If you can fix the conflict, I can merge. Nice job.

**Comment by ronyeh** [comment]
Fixed conflict. No visual diffs from master @ e7f0426a662cc9fb28760b3c547cd3a2b357e0da.

**Comment by 0xfe** [comment]
Nice job!

**Pull request merged**